content/learn/book/migration-guides/0.7-0.8/_index.md

```diff
@@ -0,0 +1,399 @@
+            // Spinning for 'duration', simulating doing hard
+        }
+        Vec2::ZERO
+    });
+    commands.spawn().insert(ComputeVec2(task))
+}
+```
+
+### [Split time functionality into bevy_time](https://github.com/bevyengine/bevy/pull/4187)
+
+* Time related types (e.g. `Time`, `Timer`, `Stopwatch`, `FixedTimestep`, etc.) should be imported from `bevy::time::*` rather than `bevy::core::*`.
+* If you were adding `CorePlugin` manually, you'll also want to add `TimePlugin` from `bevy::time`.
+* The `bevy::core::CorePlugin::Time` system label is replaced with `bevy::time::TimeSystem`.
+
+### [Move float_ord from bevy_core to bevy_utils](https://github.com/bevyengine/bevy/pull/4189)
+
+Replace imports of `bevy::core::FloatOrd` with `bevy::utils::FloatOrd`.
+
+### [Move Rect to bevy_ui and rename it to UiRect](https://github.com/bevyengine/bevy/pull/4276)
+
+The `Rect` type got renamed to `UiRect`. To migrate you just have to change every occurrence of `Rect` to `UiRect`.
+
+### [Rename ElementState to ButtonState](https://github.com/bevyengine/bevy/pull/4314)
+
+The `ElementState` type received a rename and is now called `ButtonState`. To migrate you just have to change every occurrence of `ElementState` to `ButtonState`.
+
+### [Improve docs and naming for RawWindowHandle functionality](https://github.com/bevyengine/bevy/pull/4335)
+
+rename `HasRawWindowHandleWrapper` to `ThreadLockedRawWindowHandleWrapper`
+
+### [Migrate to encase from crevice](https://github.com/bevyengine/bevy/pull/4339)
+
+#### StorageBuffer
+
+* removed set_body(), values(), values_mut(), clear(), push(), append()
+* added set(), get(), get_mut()
+
+#### UniformVec -> UniformBuffer
+
+* renamed uniform_buffer() to buffer()
+* removed len(), is_empty(), capacity(), push(), reserve(), clear(), values()
+* added set(), get()
+
+#### DynamicUniformVec -> DynamicUniformBuffer
+
+* renamed uniform_buffer() to buffer()
+* removed capacity(), reserve()
+
+### [Make paused timers update just_finished on tick](https://github.com/bevyengine/bevy/pull/4445)
+
+`Timer::times_finished has` been renamed to `Timer::times_finished_this_tick` for clarity.
+
+### [Change default Image FilterMode to Linear](https://github.com/bevyengine/bevy/pull/4465)
+
+```rs
+// 0.7
+
+//TODO
+
+// 0.8
+
+// TODO
+```
+
+### [Remove .system()](https://github.com/bevyengine/bevy/pull/4499)
+
+You can no longer use `.system()`. It was deprecated in 0.7.0. You can just remove the method call.
+
+If you needed this for tests purposes, you can use `bevy_ecs::system::assert_is_system` instead.
+
+### [Change gamepad.rs tuples to normal structs](https://github.com/bevyengine/bevy/pull/4519)
+
+The `Gamepad`, `GamepadButton`, `GamepadAxis`, `GamepadEvent` and `GamepadEventRaw` types are now normal structs instead of tuple structs and have a new() function. To migrate change every instantiation to use the `new()` function instead and use the appropriate field names instead of .0 and .1.
```

Comment:
```suggestion
The `Gamepad`, `GamepadButton`, `GamepadAxis`, `GamepadEvent` and `GamepadEventRaw` types are now normal structs instead of tuple structs and have a new() function.
To migrate change every instantiation to use the `new()` function instead and use the appropriate field names instead of .0 and .1.
```